### PR TITLE
Fixed the logic behind editing a problem resource.

### PR DESCRIPTION
**Pull request by deyordanov**
Closes https://github.com/SoftUni-Internal/exam-systems-issues/issues/1453

### CODE REVIEW

File: Services/Administration/OJS.Services.Administration.Business/ProblemResources/ProblemResourceBusinessService.cs [hunk 1]
```diff
@@ -47,13 +48,38 @@ public override async Task<ProblemResourceAdministrationModel> Edit(ProblemResou
...
+        * is not 'Link' (i.e., we're not switching from a file-based resource to a link), we want to retain the file
+        * and avoid deleting or nullifying it.
+        */
+        if (shouldKeepFile)
+        {
+            var entry = this.problemResourcesDataService.GetEntry(resource);
+
+            entry.Property(pr => pr.File).IsModified = false;
+            entry.Property(pr => pr.FileExtension).IsModified = false;
+        }
```
Comment (gogo4ds) [inline Services/Administration/OJS.Services.Administration.Business/ProblemResources/ProblemResourceBusinessService.cs:81]: Don't we need the same logic in the `Create` as well?

File: Services/Administration/OJS.Services.Administration.Models/ProblemResources/ProblemResourceAdministrationModel.cs [hunk 2]
```diff
@@ -47,6 +48,19 @@ public void RegisterMappings(IProfileExpression configuration)
             .ForMember(pr => pr.ModifiedOn, opt
                 => opt.Ignore())
             .ForMember(pr => pr.Problem, opt
-                => opt.Ignore());
+                => opt.Ignore())
+            .AfterMap((pram, pr) =>
```
Comment (gogo4ds) [inline Services/Administration/OJS.Services.Administration.Models/ProblemResources/ProblemResourceAdministrationModel.cs:52]: `pram` and `pr` are a little bit unclear. Maybe `source` and `destination`?

File: Services/Common/OJS.Services.Common.Data/IDataService.cs [hunk 3]
```diff
@@ -26,6 +27,8 @@ public interface IDataService<TEntity> : IService
 
         void Detach(TEntity entity);
 
+        public EntityEntry<TEntity> GetEntry(TEntity entity);
```
Comment (gogo4ds) [inline Services/Common/OJS.Services.Common.Data/IDataService.cs:30]: Remove the `public`

**Review by gogo4ds — CHANGES_REQUESTED** [review]

File: Services/Administration/OJS.Services.Administration.Business/ProblemResources/ProblemResourceBusinessService.cs [hunk 4]
```diff
@@ -47,13 +48,38 @@ public override async Task<ProblemResourceAdministrationModel> Edit(ProblemResou
             .Include(pr => pr.Problem)
             .FirstOrDefaultAsync();
 
+        var areFileAndLinkNull = model is { File: null, Link: null };
+
+        if (resource is { File: null } && areFileAndLinkNull)
+        {
```
Comment (gogo4ds) [inline Services/Administration/OJS.Services.Administration.Business/ProblemResources/ProblemResourceBusinessService.cs:54]: The `resource is null` check should come before this one IMO

Comment (gogo4ds) [inline Services/Administration/OJS.Services.Administration.Business/ProblemResources/ProblemResourceBusinessService.cs:54]: This check seems a bit redundant.. if file and link are null, there is no way of returning something other that true here? So I think only `areFileAndLinkNull` boolean is needed (without the `resource is { File: null }`)

**Review by gogo4ds — CHANGES_REQUESTED** [review]

File: Services/Administration/OJS.Services.Administration.Business/ProblemResources/ProblemResourceBusinessService.cs (hunk 4 above)
Comment (deyordanov) [reply Services/Administration/OJS.Services.Administration.Business/ProblemResources/ProblemResourceBusinessService.cs:54]: If ```resource is { File: null }``` is removed we would not be able to switch from one file-based resource type to another. We would throw the ```BusinessServiceException``` inside of this ```if``` check, if we  want to switch from ```AuthorsSolution``` to ```ProblemDescription``` without ```removing``` the file.

File: Services/Administration/OJS.Services.Administration.Models/ProblemResources/ProblemResourceAdministrationModel.cs (hunk 2 above)
Comment (deyordanov) [reply Services/Administration/OJS.Services.Administration.Models/ProblemResources/ProblemResourceAdministrationModel.cs]: Done.

File: Services/Common/OJS.Services.Common.Data/IDataService.cs (hunk 3 above)
Comment (deyordanov) [reply Services/Common/OJS.Services.Common.Data/IDataService.cs]: Done.

File: Services/Administration/OJS.Services.Administration.Business/ProblemResources/ProblemResourceBusinessService.cs (hunk 4 above)
Comment (deyordanov) [reply Services/Administration/OJS.Services.Administration.Business/ProblemResources/ProblemResourceBusinessService.cs:59]: Done.

**Review by gogo4ds — APPROVED** [review]

**Pull request merged**